api/genomics/src/main/java/org/pmiops/workbench/genomics/RandomizeVcf.java

```diff
@@ -0,0 +1,121 @@
+  protected File outputVcf;
+
+  @Argument(doc = "Sample name suffix.",
+      fullName = "SAMPLE_NAME_SUFFIX",
+      shortName = "S"
+  )
+  protected static String sampleNameSuffix;
+
+  static Random random = new Random();
+
+  private VariantContextWriter vcfWriter;
+
+  @Override
+  public void apply(VariantContext variant, ReadsContext readsContext,
+      ReferenceContext referenceContext, FeatureContext featureContext) {
+    vcfWriter.add(randomizeVariant(variant));
+  }
+
+  @Override
+  public void onTraversalStart() {
+    final VCFHeader inputHeader = getHeaderForVariants();
+    final List<String> newSampleNames = inputHeader.getSampleNamesInOrder().stream()
+        .map(RandomizeVcf::appendSuffixToSampleName)
+        .collect(Collectors.toList());
+    final VCFHeader outputHeader = new VCFHeader(inputHeader.getMetaDataInInputOrder(), newSampleNames);
+    vcfWriter = this.createVCFWriter(outputVcf);
+    vcfWriter.writeHeader(outputHeader);
+  }
+
+  @Override
+  public void closeTool() {
+    if ( vcfWriter != null ) {
+      vcfWriter.close();
+    }
+  }
+
+  @VisibleForTesting
+  protected static VariantContext randomizeVariant(VariantContext variant) {
+    // This initializes most of the VariantContextBuilder fields to what they were in the original variant.
+    // We just want to change the alleles, genotypes, and quality score.
+    VariantContextBuilder variantContextBuilder = new VariantContextBuilder(variant);
+    variantContextBuilder.alleles(variant.getAlleles());
+
+    List<Genotype> randomizedGenotypes = variant.getGenotypes()
+        .stream()
+        .map(genotype -> randomizeGenotype(variant, genotype))
+        .collect(Collectors.toList());
+    GenotypesContext randomizedGenotypesContext = GenotypesContext.create(new ArrayList<>(randomizedGenotypes));
+
+    variantContextBuilder.genotypes(randomizedGenotypesContext);
+
+    // We want kind of random error. If there's no error, have no error for the new variant as well.
+    // If there's error, fuzz the error.
+    if (variant.hasLog10PError()) {
+      variantContextBuilder.log10PError(random.nextDouble() * variant.getLog10PError());
+    }
+
+    return variantContextBuilder.make();
+  }
+
+  protected static Genotype randomizeGenotype(VariantContext variantContext, Genotype genotype) {
+    GenotypeBuilder genotypeBuilder = new GenotypeBuilder();
+    genotypeBuilder.copy(genotype);
+    genotypeBuilder.name(appendSuffixToSampleName(genotype.getSampleName()));
+    genotypeBuilder.alleles(randomizeAlleles(variantContext, genotype.getAlleles()));
+    return genotypeBuilder.make();
+  }
+
+  @VisibleForTesting
+  protected static List<Allele> randomizeAlleles(VariantContext variantContext, List<Allele> genotypeAlleles) {
+    // The alleles list on the VariantContext has first the reference and then all possible alternates.
+    // For each genotype, we pick from among those possible alternates (or we put a no-call.)
```

Comment:
opt: another way we could do this would be to take the global rate of no-call in the VCF file (just look it up manually and hardcode it here), then pick no call with the corresponding probability. This would probably be better since the current approach will result in strictly fewer no-calls w.r.t. the original VCF, and only in the same positions.

I was curious so I looked at the frequencies of each genotype in the sample VCF:
```
  27629 ./.
1518493 0/0
 106755 0/1
 117349 1/0
 138615 1/1
   1123 2/2
```

Perhaps this randomizer should also favor homozygous reference as well.